From ecc2b91016154d5f03d34789523c90cd4679f81d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Corentin=20No=C3=ABl?= <corentin@elementary.io>
Date: Thu, 11 Jun 2020 22:19:41 +0000
Subject: [PATCH 1/3] window-actor-x11: switch the signal callback argument and
 data on shadow factory change

invalidate_shadow takes the MetaWindowActorX11 as argument and not the MetaShadowFactory.

https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/1314


(cherry picked from commit e09ac6769809c39a055ca24be2f027affb31e370)
---
 src/compositor/meta-window-actor-x11.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/compositor/meta-window-actor-x11.c b/src/compositor/meta-window-actor-x11.c
index 53ae6ad68a..1b593280bb 100644
--- a/src/compositor/meta-window-actor-x11.c
+++ b/src/compositor/meta-window-actor-x11.c
@@ -1630,8 +1630,8 @@ meta_window_actor_x11_init (MetaWindowActorX11 *self)
 
   self->shadow_factory = meta_shadow_factory_get_default ();
   self->shadow_factory_changed_handler_id =
-    g_signal_connect (self->shadow_factory,
-                      "changed",
-                      G_CALLBACK (invalidate_shadow),
-                      self);
+    g_signal_connect_swapped (self->shadow_factory,
+                              "changed",
+                              G_CALLBACK (invalidate_shadow),
+                              self);
 }
-- 
2.26.2

